python_modules/dagit/dagit/schema/model.py

```diff
@@ -409,77 +410,98 @@ def _execution_plan_or_error(subplan_execution_args, dauphin_pipeline, evaluate_
         return EitherValue(execution_plan)
 
 
-def _execute_subplan_or_error(
-    subplan_execution_args, dauphin_pipeline, execution_plan, evaluate_value_result
-):
-    check.inst_param(subplan_execution_args, 'subplan_execution_args', SubplanExecutionArgs)
+def _execute_subplan_or_error(args, dauphin_pipeline, execution_plan, evaluate_value_result):
+    check.inst_param(args, 'args', SubplanExecutionArgs)
     check.inst_param(dauphin_pipeline, 'dauphin_pipeline', DauphinPipeline)
     check.inst_param(execution_plan, 'execution_plan', ExecutionPlan)
     check.inst_param(evaluate_value_result, 'evaluate_value_result', EvaluateValueResult)
 
-    inputs_to_marshal = defaultdict(dict)
-    outputs_to_marshal = defaultdict(list)
-
-    for step_execution in subplan_execution_args.step_executions:
-        step = execution_plan.get_step_by_key(step_execution.step_key)
-        for marshalled_input in step_execution.marshalled_inputs:
-            if not step.has_step_input(marshalled_input.input_name):
-                schema = subplan_execution_args.info.schema
-                return EitherError(
-                    schema.type_named('StartSubplanExecutionInvalidInputError')(
-                        step=schema.type_named('ExecutionStep')(step),
-                        invalid_input_name=marshalled_input.input_name,
-                    )
-                )
-
-            inputs_to_marshal[step.key][marshalled_input.input_name] = marshalled_input.key
-
-        for marshalled_output in step_execution.marshalled_outputs:
-            if not step.has_step_output(marshalled_output.output_name):
-                schema = subplan_execution_args.info.schema
-                return EitherError(
-                    schema.type_named('StartSubplanExecutionInvalidOutputError')(
-                        step=schema.type_named('ExecutionStep')(step),
-                        invalid_output_name=marshalled_output.output_name,
-                    )
-                )
-
-            outputs_to_marshal[step.key].append(
-                {'output': marshalled_output.output_name, 'path': marshalled_output.key}
-            )
-
-    schema = subplan_execution_args.info.schema
+    # convenient shortcuts for brevity
+    type_of = args.info.schema.type_named
```

Comment:
these are still a little gnomic and it would be nice to have a little more comment explaining their intended usage. in the diff it's clear what they're doing but when someone comes along to the new code (esp without type hints), stuff like this is going to be a little weird:

```type_of('InvalidSubplanExecutionError')(step=schema.type_named('ExecutionStep')(step=type_of('ExecutionStep')(step_of(invalid_subplan_error))```

Reply:
excellent use of gnomic